RPLCD/gpio.py

```diff
@@ -85,8 +87,16 @@ def __init__(self, numbering_mode=None, pin_rs=None, pin_rw=None, pin_e=None, pi
         :param auto_linebreaks: Whether or not to automatically insert line
             breaks. Default: ``True``.
         :type auto_linebreaks: bool
+        :param legacy_mode: Whether to run additional checks to support older LCDs
+            that may not run at the reference clock (or keep up with it).
+        :type legacy_mode: bool
 
         """
+        # Configure legacy mode
+        self.legacy_mode = legacy_mode
+        if legacy_mode:
+            self.last_send_event = time.time()
```

Comment:
I think it would be better to use [time.monotonic](https://docs.python.org/3.5/library/time.html#time.monotonic) since you want to measure time elapsed. Otherwise an NTP update during a write could have unintended consequences.

However, it's only available on Python 3.3+ and we currently still support 2.7 (which I'd like to continue until support ends). What about a helper function that uses `time.monotonic` if available, and falls back to `time.time`?

Reply:
You could also use this:

```
if sys.version_info.major < 3:
    from time import time as now
else:
    from time import monotonic as now
```

Reply:
+2 for monotonic.

Reply:
That's a good point, my mistake for not noticing that. I do think that using `perf_counter` (3+) and `clock` (2.7) would be better, since those have a higher resolution; I made changes based on that. Let me know what you think.

Reply:
> since those have a higher resolution

do they? did you notice better performance?

Reply:
I did not notice anything perceptible, no. From what I can tell, `perf_counter()` returns the highest resolution timer on the OS (I ran the code on Windows), and I was getting much greater accuracy with that over `monotonic()`. This could be different on the RPi, though.